include/picongpu/particles/debyeLength/Check.kernel

```diff
@@ -0,0 +1,227 @@
+                        // Supercells with no particles do not contribute to Debye length estimate
+                        if(supercell.getNumParticles())
+                            process(acc, electronBox, supercellIdx, result);
+                    }
+
+                private:
+                    /** Check Debye length for electrons in the given supercell
+                     *
+                     * Ths supercell is required to have macroparticles
+                     *
+                     * @tparam T_Acc alpaka accelerator type
+                     * @tparam T_ElectronBox electron storage type
+                     * @tparam T_ResultBox result storage type, device box for Result
+                     *
+                     * @param acc alpaka accelerator
+                     * @param electronBox electron storage
+                     * @param supercellIdx supercell index, including guards
+                     * @param result device box of type Result with a single element
+                     *        to write kernel result to
+                     */
+                    template<typename T_Acc, typename T_ElectronBox, typename T_ResultBox>
+                    DINLINE void process(
+                        T_Acc const& acc,
+                        T_ElectronBox const electronBox,
+                        pmacc::DataSpace<simDim> const supercellIdx,
+                        T_ResultBox result) const
+                    {
+                        using namespace pmacc::mappings::threads;
+                        constexpr uint32_t frameSize = pmacc::math::CT::volume<SuperCellSize>::type::value;
+                        constexpr uint32_t numWorkers = T_numWorkers;
+                        using ParticleDomCfg = IdxConfig<frameSize, numWorkers>;
+                        // Use double-precision for better accuracy, this kernel is not performance-critical
+                        auto sumMomentum = float3_64::create(0.0);
+                        auto sumMomentumSquared = float3_64::create(0.0);
+                        auto frame = electronBox.getLastFrame(supercellIdx);
+                        auto particlesInFrame = electronBox.getSuperCell(supercellIdx).getSizeLastFrame();
+                        uint32_t const workerIdx = cupla::threadIdx(acc).x;
+                        while(frame.isValid())
+                        {
+                            // parallel loop over all particles in the frame
+                            ForEachIdx<ParticleDomCfg>{workerIdx}([&](uint32_t const linearIdx, uint32_t const) {
+                                if(linearIdx < particlesInFrame)
+                                {
+                                    auto const particle = frame[linearIdx];
+                                    // the evaluation is based on single- (not macro-) particle momentums
+                                    auto const singleParticleMomentum = precisionCast<float_64>(particle[momentum_])
```

Comment:
Why do you cast first to 64 bit instead of deciding first and then cast to 64bit?

Reply:
So that all operations are in 64 bit. Although in this case it should not really matter.